fastparquet/writer.py

```diff
@@ -893,6 +910,24 @@ def write(filename, data, row_group_offsets=50000000,
         If False, construct data-set from scratch; if True, add new row-group(s)
         to existing data-set. In the latter case, the data-set must exist,
         and the schema must match the input data.
+    date_col: str (None)
+        Name of the column to use for grouping by date in case of a date row
+        grouping. If `None`, index is used.
+    drop_duplicates_on: str, list of str or None (None)
+        Parameter used in case of a date row grouping.
+        Either `None`, or a str or list of str specifying names of the columns
+        to be used to identify duplicates.
+          * if a str or list of str, date column (or index) for grouping is
+            necessarily added to this list. It can be only the name of the
+            column/index embedding the dates to restrict to this data.
+          * if `None`, both date column (or index) for grouping and all columns
+            are used to identify duplicates.
+        Dropping duplicates on column values without the date used for grouping
```

Comment:
I am not sure we understand each other with the word 'defer'. You mean, you want to have this parameter generic for all append situations, even without date row grouping, right?

> Is it essential for your use case?
I confirm it is.
I introduced my use case in my above answer.
I need this for data reconciliation.
I have data streams that produce continuously 24h/7d data, with sometime redundant data.
I am merging this data in a single data store.
With `drop_duplicates_on` parameter, the fastparquet `write` function makes this reconciliation job transparently, and thanks to `date_row_grouping`, it makes it as efficiently as I think it is feasible because thanks to the dates, it localies the data to reconcile, and overwrite only the files that are impacted.